src/classes/BDI_MigrationMappingUtility.cls

```diff
@@ -127,6 +121,23 @@ public class BDI_MigrationMappingUtility {
         FROM Data_Import_Field_Mapping_Set__mdt
         WHERE DeveloperName = :DEFAULT_FIELD_MAPPING_SET_RECORD_NAME];
 
+    /**
+    * @description Boolean flag to determine if a custom field mapping set exists 
+    */
+    public Boolean customFieldMappingSetExists {
```

Comment:
Are you still using this method?

Reply:
Yep, still using this in a couple of places. One at the beginning of a migration to check whether or not we're going to be comparing against the default field set or the custom field set and when we're updating the data import custom settings (my thinking here is that when Advanced Mapping is turned on through the migration utility, we always point to the custom field mapping set if it exists).

Reply:
Oh right - not a method, variable with a getter. Duh.

Reply:
I notice below you have a local variable with the same name that just checks the list size > 0 to set the boolean - I just want to be sure a) we don't get tripped up with the two variables having the same name and b) aren't querying twice, once to see if it exists and again to pull the value.

Reply:
100%. Looking at this again what do you think of the following...

Right now I have both a class and local var because upsertCustomSettings is static. I use it in BDI_MigrationMappingUtility, DeploymentCallback, and STG_PanelDataImportAdvancedMapping_CTRL (vfp controller). Doesn't really make sense now but I think it must've at point before the changes.

I think I'd prefer to unstatic-fy upsertCustomSettings, toss out that local var, and use the class var. I'll just create a new instance of BDI_MigrationMappingUtility in the DeploymentCallback class where I need to call upsertCustomSettings. Not needed in STG_PanelDataImportAdvancedMapping_CTRL since already creating an instance of BDI_MigrationMappingUtility in the constructor and can use that.

Reply:
I think this totally makes sense - I'm guessing that the DeploymentCallback was probably operating in a new static context anyway, since it's asynchronous. Let me know if you run into hitches building that out.